cmd/podman/common.go

```diff
@@ -137,6 +137,10 @@ var createFlags = []cli.Flag{
 		Name:  "cpuset-mems",
 		Usage: "Memory nodes (MEMs) in which to allow execution (0-3, 0,1). Only effective on NUMA systems.",
 	},
+	cli.StringSliceFlag{
+		Name:  "depends-on",
+		Usage: "Add dependency containers which must be running before this container starts",
```

Comment:
How is a human going to know this?  Not sure what this even means?  I am not sure we want to expose this or support it, maybe should be hidden?

Reply:
I was mostly thinking of ordering service startup - ensuring that the DB container is started before the HTTP server container spins up. This isn't good microservices practice, but I can see this as an easy way establishing service ordering without getting into systemd unit files.

Reply:
Those are jobs for systemd not for podman.
I could see this patch set becoming very complex and ordering is notoriously bad.  You would have to worry about circular dependencies.  ETC. I don't think this is a good idea.  Leave it to Systemd.

Reply:
We already have to worry about those things for namespace dependencies - that's why we have the dependency graph generation code we use for pod startup.

Reply:
Sure, but that is the definition of the pod, but I don't see this as adding anything but headaches, and would be something we would want to discourage.  As soon as you get into complex dependencies, like this there are lots of errors.  

I don't see us advising users to use this, we would advise them to setup services to run in unit files with proper dependencies defined in systemd. Then they would always be defining them in the same place.

Reply:
If you want, we can restrict it to only be used within pods at the CLI level, but the underlying code is generic enough that we can use it without pods if we want to. Within pods, the typical example would be to depend on the pause container in Kube pods, but I'm sure people can think of other ways of using this.

Reply:
Sure I can think of ways people would use it, but this is not something we want to support.

Reply:
If you want, I can scrap the CLI frontend. The backend code is sufficiently trivial/useful that I'd like to leave it in. I also think start/remove --recursive are useful enough to justify (`podman remove --recursive --force` is needed to guarantee we kick out a container right now)

Reply:
Remove the CLI, and then I am less concerned.  If we find the need we could eventually add it back.
Not sure you need --recursive.  Perhaps --force should be enough.  I am not sure what docker does in this situation.  Does it fail to stop a container if a second container relies on it, or does it kill the first and the second keeps open the namespaces that are shared.   

podman rm --force
CID=podman run -d ...
podman run -d --net  container:$CID --PID container $CID ...
podman stop $CID
What happens?

podman rm $CID --force

What happens?